File: Backend/API/src/main/java/com/example/localhistory/user/dto/response/StudentLeaderboardDTO.java
package com.example.localhistory.user.dto.response;

import lombok.Data;

// Response row for the global student leaderboard.
@Data
public class StudentLeaderboardDTO {
    private Integer rank;
    private Long studentId;
    private String firstName;
    private String lastName;
    private String email;
    private Integer level;
    private Integer points;
    private Integer pointsRequired;
    private Boolean currentUser;
}

File: Frontend/app/src/main/java/com/example/localhistory/model/response/StudentLeaderboardDTO.kt
package com.example.localhistory.model.response

data class StudentLeaderboardDTO(
    val rank: Int,
    val studentId: Long,
    val firstName: String,
    val lastName: String,
    val email: String,
    val level: Int,
    val points: Int,
    val pointsRequired: Int,
    val currentUser: Boolean
)
